### PR TITLE
Add "Used by" section to README

### PR DESCRIPTION
**Pull request by MatthiasSchmidtblaicherQC**
In addition to tabmat and glum, I linked to linearmodels. But I am sure that I missed other projects, so please, feel free to expand the list!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ import pandas
 from formulaic import Formula
 
 df = pandas.DataFrame({
-    'y': [0,1,2],
+    'y': [0, 1, 2],
```
Comment (MatthiasSchmidtblaicherQC) [inline README.md:41]: I know, this change is very pedantic, but I could not help adding the spaces.

**Comment by s3alfisc** [comment]
Not sure if it already merits to be in the company of the excellent packages listed above - but [pyfixest](https://github.com/s3alfisc/pyfixest) also heavily builds on `formulaic`. My feeling is that listing `pyfixest` would do more in terms of promoting `pyfixest` than `formulaic` at its current stage =)

**Comment by s3alfisc** [comment]
Thanks! =)

**Comment by matthewwardrop** [comment]
Apologies for the delay on this one! Thanks @MatthiasSchmidtblaicherQC for putting this together :).

**Pull request merged**